### PR TITLE
fix load jni lib failed

### PR DESCRIPTION
**Pull request by jackylee-ch**
### What changes were proposed in this pull request?
Fix load jni lib while running with spark speculation.
`JniLoader.get().ensureLoaded()` maybe failed with `ClosedByInterruptException` as the task is killed. It would always failed  to init `NativeMemoryPool` with `java.lang.NoClassDefFoundError: Could not initialize class org.apache.arrow.dataset.jni.NativeMemoryPool`

### How was this patch tested?
unit tests

### CODE REVIEW

File: java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeMemoryPool.java [hunk 1]
```diff
@@ -35,6 +35,7 @@ private NativeMemoryPool(long nativeInstanceId) {
    * Get the default memory pool. This will return arrow::default_memory_pool() directly.
    */
   public static NativeMemoryPool getDefault() {
+    ensureJniLoaded()
```
Comment (weiting-chen) [inline java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeMemoryPool.java:38]: missing ; at the end?

File: java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeMemoryPool.java [hunk 2]
```diff
@@ -44,6 +45,7 @@ public static NativeMemoryPool getDefault() {
    * from the listener in advance.
    */
   public static NativeMemoryPool createListenable(ReservationListener listener) {
+    ensureJniLoaded()
```
Comment (weiting-chen) [inline java/dataset/src/main/java/org/apache/arrow/dataset/jni/NativeMemoryPool.java:48]: missing ; at the end?

**Comment by zhztheplayer** [comment]
@jackylee-ch If you can try contributing the change to Apache Arrow?

**Comment by jackylee-ch** [comment]
> @jackylee-ch If you can try contributing the change to Apache Arrow?

Sure.

**Pull request merged**